arch/arc/include/asm/atomic64-arcv2.h

```diff
@@ -53,29 +53,29 @@ static inline void atomic64_##op(s64 a, atomic64_t *v)			\
 									\
 	__asm__ __volatile__(						\
 	"1:				\n"				\
-	"	llockd  %0, [%1]	\n"				\
-	"	" #op1 " %L0, %L0, %L2	\n"				\
-	"	" #op2 " %H0, %H0, %H2	\n"				\
-	"	scondd   %0, [%1]	\n"				\
+	"	llockd  %0, [%2]	\n"				\
+	"	" #op1 " %L0, %L0, %L3	\n"				\
+	"	" #op2 " %H0, %H0, %H3	\n"				\
+	"	scondd   %0, [%2]	\n"				\
 	"	bnz     1b		\n"				\
-	: "=&r"(val)							\
+	: "=&r"(val), "+m"(v->counter)					\
 	: "r"(&v->counter), "ir"(a)					\
 	: "cc");							\
 }									\
 
-#define ATOMIC64_OP_RETURN(op, op1, op2)		        	\
+#define ATOMIC64_OP_RETURN(op, op1, op2)				\
 static inline s64 atomic64_##op##_return_relaxed(s64 a, atomic64_t *v)	\
 {									\
 	s64 val;							\
 									\
 	__asm__ __volatile__(						\
 	"1:				\n"				\
-	"	llockd   %0, [%1]	\n"				\
-	"	" #op1 " %L0, %L0, %L2	\n"				\
-	"	" #op2 " %H0, %H0, %H2	\n"				\
-	"	scondd   %0, [%1]	\n"				\
+	"	llockd   %0, [%2]	\n"				\
+	"	" #op1 " %L0, %L0, %L3	\n"				\
+	"	" #op2 " %H0, %H0, %H3	\n"				\
+	"	scondd   %0, [%2]	\n"				\
 	"	bnz     1b		\n"				\
-	: [val] "=&r"(val)						\
+	: "=&r"(val), "+m"(v->counter)					\
```

Comment:
Likewise. I cannot understand what you do with second argument `%1`) and why you need the third argument (`%2`) when you have the second one.

Reply:
Done